### DISCUSSION TITLE
General question about serverless function payload size limit

### DISCUSSION THREAD

**Discussion by mikeamarkey** [vercel/vercel#4431] [General] [answered]
So, I know that serverless function request/response payloads are limited to 5mb. But within the function itself, we are free to make requests/receive responses from other endpoints that might be greater than 5mb.

I was wondering why this is the case. Is there something about the technology of serverless functions themselves that necessitates a smaller request/response payload?

**Comment by paulogdm** [comment, marked as answer]
Hello,

This is an upstream provider limitation:

1. https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html
2. https://stackoverflow.com/questions/46298060/aws-lambda-response-error

The ideal solution would be to simply have the client do the upload to an S3 bucket. This is outdated but you can grab the idea from it: https://github.com/paulogdm/s3-example

**Reply by ZoranRavic** [reply]
The amazon docs that you linked say that the limit is 6 MB, while the Vercel docs currently say 4.5 MB.

Also on Amazon they say that they "use the abbreviation MB (rather than MiB) to refer to 1024 KB" while Vercel does not say whether their limits are in MB or MiB.

**Reply by leerob** [reply]
The Vercel documentation is correct - the current limit is 4.5 MB. 

> I noticed that even 3.5 MB fails with code 413

Can you share the code?

**Reply by ZoranRavic** [reply]
> > I noticed that even 3.5 MB fails with code 413
> 
> Can you share the code?

I made a separate test and confirmed that 4.5 MB works.
I can't re-test the code where I saw the problem, it's on a production server and relies on data that has changed, but if I find something reproducible I'll file a report.

**Comment by mikeamarkey** [comment]
@paulogdm thanks for the reply.

Got it. Yeah, we're going to do something like with the user uploading directly to the bucket.

I'm still wondering S3's reasoning behind the limit. The fact that the limit can not be increased makes me think there is some special reasoning behind the choice to limit the size. Although Azure functions apparently allow up to 100mb for request body, so maybe it's just Amazon's implementation.

Anyways, I see that it's not related to vercel. Thanks again 👋
